### ISSUE TITLE
Incompatibility with Rubygems starting from 3.0.0.beta2

### ISSUE THREAD

**Issue by sponomarev** [oracle/truffleruby#1558]
```sh

$ ruby -v
truffleruby 1.0.0-rc11, like ruby 2.4.4, GraalVM CE Native [x86_64-darwin]

$ gem -v
2.6.14.1

$ bundle -v
Bundler version 2.0.1

$ gem update --system
Updating rubygems-update
Fetching: rubygems-update-3.0.2.gem (100%)
Successfully installed rubygems-update-3.0.2
Installing RubyGems 3.0.2
Bundler 1.17.3 installed
RubyGems 3.0.2 installed
Regenerating binstubs
ERROR:  While executing gem ... (RuntimeError)
    can't modify frozen Hash

$ gem -v
3.0.2

 $ bundle -v
/Users/toad/.rbenv/versions/truffleruby-1.0.0-rc11/lib/patches/rubygems/request_set/gem_dependency_api.rb:10:in `[]=': can't modify frozen Hash (RuntimeError)
	from /Users/toad/.rbenv/versions/truffleruby-1.0.0-rc11/lib/patches/rubygems/request_set/gem_dependency_api.rb:10:in `GemDependencyAPI'
	from /Users/toad/.rbenv/versions/truffleruby-1.0.0-rc11/lib/patches/rubygems/request_set/gem_dependency_api.rb:3:in `<top (required)>'
	from /Users/toad/.rbenv/versions/truffleruby-1.0.0-rc11/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:54:in `gem_original_require'
	from /Users/toad/.rbenv/versions/truffleruby-1.0.0-rc11/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Users/toad/.rbenv/versions/truffleruby-1.0.0-rc11/lib/ruby/site_ruby/2.4.0/rubygems/request_set.rb:477:in `<top (required)>'
	from /Users/toad/.rbenv/versions/truffleruby-1.0.0-rc11/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:54:in `gem_original_require'
	from /Users/toad/.rbenv/versions/truffleruby-1.0.0-rc11/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Users/toad/.rbenv/versions/truffleruby-1.0.0-rc11/lib/ruby/site_ruby/2.4.0/rubygems.rb:238:in `finish_resolve'
	from /Users/toad/.rbenv/versions/truffleruby-1.0.0-rc11/lib/ruby/site_ruby/2.4.0/rubygems.rb:305:in `block in activate_bin_path'
	from /Users/toad/.rbenv/versions/truffleruby-1.0.0-rc11/lib/ruby/site_ruby/2.4.0/rubygems.rb:303:in `synchronize'
	from /Users/toad/.rbenv/versions/truffleruby-1.0.0-rc11/lib/ruby/site_ruby/2.4.0/rubygems.rb:303:in `activate_bin_path'
	from /Users/toad/.rbenv/versions/truffleruby-1.0.0-rc11/bin/bundle:23:in `<main>'
```

**Comment by sponomarev** [comment]
Broken since https://github.com/rubygems/rubygems/commit/0a0663ff6a341c62edbaf5c3143024f3bef04076

**Comment by eregon** [comment]
Thank you for the report!
I guess we should just upstream these patches to RubyGems, so there is no need for them in the first place.

**Comment by chrisseaton** [comment]
Fixed in 6f1b9aa973ee8d81f8eb56809d66fb7ea26e2b3a.
